tmt/utils/__init__.py

```diff
@@ -6280,3 +6281,33 @@ def is_url(url: str) -> bool:
     """ Check if the given string is a valid URL """
     parsed = urllib.parse.urlparse(url)
     return bool(parsed.scheme and parsed.netloc)
+
+
+# Handle the thread synchronization for the `catch_warnings(...)` context manager
+_catch_warning_lock = RLock()
+ActionType = Literal['default', 'error', 'ignore', 'always', 'all', 'module', 'once']
+
+
+@contextlib.contextmanager
+def catch_warnings(
```

Comment:
`suppress_warnings` seemed better for me, as we don't "catch" those warnings, as in "try -> except -> do something about the error", we suppress them.

Reply:
Ok, makes sense. I missed we're allowing more actions.